strategies/affected/src/main/java/org/arquillian/smart/testing/strategies/affected/ClassDependenciesGraph.java

```diff
@@ -73,14 +79,86 @@ void buildTestDependencyGraph(Collection<File> testJavaFiles) {
 
         // Then find dependencies
         for (String changedTestClassNames : testClassesNames) {
-            JavaClass javaClass = builder.getClassDescription(changedTestClassNames);
-            if (javaClass != null) {
-                addToIndex(new JavaElement(javaClass), javaClass.getImports());
+            JavaClass testJavaClass = builder.getClassDescription(changedTestClassNames);
+            if (testJavaClass != null) {
+                final String[] imports = testJavaClass.getImports();
+                final List<String> manualProductionClasses = calculateManualAddedDependencies(testJavaClass);
+                manualProductionClasses.addAll(Arrays.asList(imports));
+                addToIndex(new JavaElement(testJavaClass), manualProductionClasses);
             }
         }
     }
 
-    private void addToIndex(JavaElement javaElement, String[] imports) {
+    private List<String> calculateManualAddedDependencies(JavaClass testJavaClass) {
+        final List<String> manualDependencyClasses = new ArrayList<>();
+        final Tests[] allTestsAnnotation = getAllAnnotations(testJavaClass);
+
+        for (Tests tests : allTestsAnnotation) {
+            List<String> packages = getPackages(testJavaClass.packageName(), tests);
+            for (String packag : packages) {
```

Comment:
Typo `packag` :)

Reply:
Well it is not a type, `package` is a reserved word, so I decided to put in this way. But any other suggestion is welcome :)

Reply:
That was late hahahaha. `pkg` maybe? :)

![facepalm-real](https://user-images.githubusercontent.com/719616/32325299-a9522b48-bfce-11e7-83a5-e1267e236610.gif)